ocaml/xapi/network_event_loop.ml

```diff
@@ -0,0 +1,92 @@
+module D=Debug.Make(struct let name="network_event_loop" end)
+
+module StringMap = Map.Make(struct type t = string let compare = compare end)
+
+type 'a update = Deleted | Snapshot of 'a
+
+let _watch_networks __context ~update_firewall ~wait_after_failure_seconds =
+
+  (* We keep track of the network objects in the database using this event loop. *)
+  let classes = ["network"] in
+  (* We keep track of the interfaces that we last passed to the firewall script
+     to allow NBD traffic on them. At startup, we don't know on which
+     interfaces NBD is allowed, and we always update the firewall. *)
+  let allowed_interfaces = None in
+
+  let api_timeout = 60. in
+  let timeout = 30. +. api_timeout +. !Db_globs.master_connection_reset_timeout in
+
+  let wait_for_network_change ~token =
+    let from =
+      Helpers.call_api_functions ~__context
+        (fun rpc session_id ->
+           Client.Client.Event.from ~rpc ~session_id ~classes ~token ~timeout |> Event_types.event_from_of_rpc)
+        ~test_fn:(fun () -> Xapi_event.from ~__context ~classes ~token ~timeout |> Event_types.parse_event_from)
+    in
+    let events = from.Event_types.events in
+    List.iter (fun event -> D.debug "got event %s" (Event_types.string_of_event event)) events;
```

Comment:
Probably don't want to log every single event here

Reply:
Hmm ok, I'll remove it. My thinking was that this should not be triggered oftern, because we're just using events here as a notification mechanism, and query the DB. So if this is triggered too often, that will lead to many database queries and inefficiency.

Reply:
Right, if events on the network class are common, we probably have bigger problems.
I imagine that these log-lines are not that likely to be useful for diagnosing problems in real deployments though.